src/proxy_handler.cc

```diff
@@ -0,0 +1,180 @@
+#include "proxy_handler.h"
+
+#include <string>
+
+RequestHandler::Status ProxyHandler::Init(const std::string& uri_prefix,
+    const NginxConfig& config)
+{
+  uri_prefix_ = uri_prefix;
+
+  int server_location = 0;
+  if (config.statements_[0]->tokens_[0] == "remote_port") {
+    server_location = 1;
```

Comment:
Is this so that you can either specify `remote_port google.com` or just `google.com` if you wanted? Maybe leave a comment on expected format somewhere?